test/integration/node-specific/crypto_callbacks.test.ts

```diff
@@ -1,179 +0,0 @@
-import { expect } from 'chai';
-import * as sinon from 'sinon';
-
-/* eslint-disable @typescript-eslint/no-restricted-imports */
-import { ClientEncryption } from '../../../src/client-side-encryption/client_encryption';
-/* eslint-disable @typescript-eslint/no-restricted-imports */
-import * as cryptoCallbacks from '../../../src/client-side-encryption/crypto_callbacks';
-import { type MongoClient } from '../../mongodb';
-
-// Data Key Stuff
-const AWS_ACCESS_KEY_ID = process.env.AWS_ACCESS_KEY_ID;
-const AWS_SECRET_ACCESS_KEY = process.env.AWS_SECRET_ACCESS_KEY;
-const AWS_REGION = process.env.AWS_REGION;
-const AWS_CMK_ID = process.env.AWS_CMK_ID;
-
-const kmsProviders = {
-  aws: { accessKeyId: AWS_ACCESS_KEY_ID, secretAccessKey: AWS_SECRET_ACCESS_KEY }
-};
-const dataKeyOptions = { masterKey: { key: AWS_CMK_ID, region: AWS_REGION } };
-
-const SKIP_AWS_TESTS = [AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION, AWS_CMK_ID].some(
-  secret => !secret
-);
-
-describe('cryptoCallbacks', function () {
```

Comment:
I see the rationale for removing the tests since they're broken, but is there any way we can still test this functionality?

Reply:
Do we think that we have sufficient coverage for these functions in `mongodb-client-encryption`?

Reply:
Ha.  No.  We have no tests for them really in mongodb-client-encryption.

But these tests aren't any better - they also don't actually test the crypto hooks, they spy on them and assert that they're called.  Or make the spy error and assert the error is propagated to the user.

Reply:
Maybe I'm not quite understanding, how do we know that the functions work? Are we just relying on the tests that live inside the libmongocrypt itself and hoping that the bindings just work™️?

Reply:
The JS crypto callbacks are simple: https://github.com/mongodb-js/mongodb-client-encryption/blob/main/src/crypto_callbacks.ts

The C++ crypto callbacks are basically copy-pasting code from libmongocrypt, with some minor modifications: https://github.com/mongodb-js/mongodb-client-encryption/blob/main/addon/openssl-crypto.cc and https://github.com/mongodb/libmongocrypt/blob/18cb9e4e900c45c0b9b71fd34e159f8cb29fe1de/src/crypto/libcrypto.c.

They're implicitly covered with all our FLE tests.  We never invoke these functions directly - they're called from libmongocrypt when we invoke FLE operations.  Our existing FLE test suite implicitly covers the crypto hooks.

We probably could test them explicitly, although I personally think that there are diminishing returns to testing them directly.  Outside of the happy path (successfully getting an encrypted or decrypted value), testing other scenarios is involved for marginal benefit (without reworking our c++ crypto callbacks, for example, I'm not sure how to test anything but the happy path.  how do we make opensll give us the errors we expect?).